chicago95: init at 3.0.1

GabrielMaguire opened the pull request:
## Description of changes

Windows 95 theme for linux. GTK theme, icons, fonts, and an sddm theme.

Homepage: https://github.com/grassmunk/Chicago95

Question to reviewer(s): Should this package be broken up into `chicago95-{theme,icons,fonts,sddm}` or is one package acceptable?

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GabrielMaguire committed:
maintainers: add GabrielMaguire

GabrielMaguire committed:
chicago95: init at 3.0.1

Gliczy commented on pkgs/data/themes/chicago95/default.nix:
```diff
@@ -0,0 +1,50 @@
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Windows 95 theme for linux";
+    homepage = "https://github.com/grassmunk/Chicago95";
+    license = with licenses; [
+      gpl3Plus
+      mit
+    ];
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ GabrielMaguire ];
+  };
```
```suggestion
  meta = {
    description = "Windows 95 theme for linux";
    homepage = "https://github.com/grassmunk/Chicago95";
    license = with lib.licenses; [
      gpl3Plus
      mit
    ];
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ GabrielMaguire ];
  };
```

Gliczy commented on pkgs/data/themes/chicago95/default.nix:
```diff
@@ -0,0 +1,50 @@
+  '';
+
+  meta = with lib; {
+    description = "Windows 95 theme for linux";
```
```suggestion
    description = "Windows 95 theme for GTK";
```

GabrielMaguire commented:
@Gliczy, thank you for the review! I believe I have addressed the review comments.

One question, sometimes reviewers prefer a merge workflow once the review has begun, however, I'm assuming a rebase workflow here is common due to the strict/clean commit log requirements (and no squash merging). For simple PRs like this I can see it doesn't matter as much since reviewing the entire PR is relatively fast, but does the rebase workflow still hold for more complex PRs where being able to easily track diffs between reviews may be important to reviewers?

Gliczy commented:
LGTM!

Yes, it still holds. Each logical unit changes must be made in separate commits.
So, in the case of more complex PRs where it's likely to touch multiple part (and so separate commits), the diffs can be reviewed per commit.

Gliczy reviewed (approved):
Result of `nixpkgs-review pr 335968` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chicago95</li>
  </ul>
</details>

GabrielMaguire commented:
@Gliczy, do you know if there is anything else required from my end to merge this PR?

- I see other merged PRs do not require the darwin CI checks to complete if not applicable, so I think that should be good here
- From the [contributing docs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-merge-pull-requests) I see there is the mention "_Nixpkgs committers_ are people who have been given permission to merge", but I'm unclear if the _Nixpkgs committers_ is a separate group that I need to ping.
- Your approval check mark is gray instead of green as seen on other merged PRs
